synapse/http/matrixfederationclient.py

```diff
@@ -172,7 +173,14 @@ def __attrs_post_init__(self) -> None:
 
         # The object is frozen so we can pre-compute this.
         uri = urllib.parse.urlunparse(
-            (b"matrix", destination_bytes, path_bytes, None, query_bytes, b"")
+            (
+                b"matrix-federation",
```

Comment:
@erikjohnston Any background on the new `matrix-federation://` schema? Just making it more clear?

Reply:
Yeah, so `matrix://` is actually a registered scheme that does *not* mean the same thing. Since the `matrix://` scheme is no longer going to just be internal to synapse (it'll be going over the wire), we thought we should be good and change it.